pkg/datarepair/checker/checker.go

```diff
@@ -247,6 +246,20 @@ func (checker *Checker) updateSegmentStatus(ctx context.Context, pointer *pb.Poi
 	return nil
 }
 
+// IrreparableSegmentCron runs checker's Irreparable process service
+func (checker *Checker) IrreparableSegmentCron(ctx context.Context) (err error) {
+	defer mon.Task()(&ctx)(&err)
+	checker.logger.Info("checker's Irreparable process cron is starting up")
+
+	return checker.IrreparableLoop.Run(ctx, func(ctx context.Context) error {
+		err := checker.IrreparableProcess(ctx)
```

Comment:
just return the err...
`return checker.IrreparableProcess(ctx)`

Reply:
or a wrapped  version of the err

Reply:
used inline